plugins/plugin-watermark/src/node/watermarkPlugin.ts

```diff
@@ -0,0 +1,29 @@
+import type { Plugin } from 'vuepress/core'
+import { getDirname, path } from 'vuepress/utils'
+import type { WatermarkPluginOptions } from '../shared/index.js'
+
+const __dirname = getDirname(import.meta.url)
+
+export const watermarkPlugin = ({
+  pageFilter = () => true,
+  ...options
+}: WatermarkPluginOptions = {}): Plugin => {
+  options.global ??= true
```

Comment:
Add a logger for debug propose could be better as this could probably be a theme plugin.